### PR TITLE
Updated PyTorch build instructions

### PR DESCRIPTION
**Pull request by pruthvistony**

### CODE REVIEW

**Comment by pruthvistony** [comment]
@Rmalavally ,
Please review these changes.

**Comment by pruthvistony** [comment]
@sunway513 , @jeffdaily, @jithunnair-amd
Please review these changes

**Comment by Rmalavally** [comment]
I will wait to review and commit until we have the sign-off from others. Thanks, Pruthvi.

**Comment by pruthvistony** [comment]
> I will wait to review and commit until we have the sign-off from others. Thanks, Pruthvi.

Thanks.
I made these changes currently only for this issue + small delta. 
But I feel more updates/changes are required to handle few obsolete information and i hope they is plan to take care of this.

**Review by Rmalavally — APPROVED** [review]

**Comment by Rmalavally** [comment]
Please review the technical changes as requested.

**Comment by pruthvistony** [comment]
@sunway513 
As discussed, the current changes support CentOS and Ubuntu.
I have also removed gfx803.

**Review by sunway513 — APPROVED** [review]

File: Deep_learning/Deep-learning.rst [hunk 1]
```diff
@@ -372,16 +372,16 @@ This step is optional but most PyTorch scripts will use torchvision to load mode
 Option 3: Install using minimal ROCm docker file
 ************************************************
 
-1. Download pytorch dockerfile:
+1. Download dockerfile based on the OS choose:
+Recommend to use - Dockerfile-<OS distro>-complete to get all the ROCm Math libs installed which are required for PyTorch.
```
[the remark this answers is not in the document]
Comment (pruthvistony) [reply Deep_learning/Deep-learning.rst:376]: Above changes done are common for CentOS and Ubuntu.

File: Deep_learning/Deep-learning.rst [hunk 2]
```diff
@@ -407,42 +407,41 @@ Note: This will mount your host home directory on /data in the container.
...
-  python tools/amd_build/build_amd.py
+  python3 tools/amd_build/build_amd.py
 
 6. Build and install pytorch:
 
-Unless you are running a gfx900/Vega10-type GPU (MI25, Vega56, Vega64,...), explicitly export the GPU architecture to build for, e.g.:
-export HCC_AMDGPU_TARGET=gfx906
+By default pytorch is built for all supported AMD GPU targets like gfx803/gfx900/gfx906/gfx908 (MI25, MI50, MI60, MI100, ...)
+This can be overwritten using
+export PYTORCH_ROCM_ARCH=gfx803;gfx900
```
[the remark this answers is not in the document]
Comment (pruthvistony) [reply Deep_learning/Deep-learning.rst]: Done.

File: Deep_learning/Deep-learning.rst [hunk 3]
```diff
@@ -372,18 +372,18 @@ This step is optional but most PyTorch scripts will use torchvision to load mode
 Option 3: Install using minimal ROCm docker file
 ************************************************
 
-1. Download pytorch dockerfile:
+1. Download dockerfile based on the OS choose:
+Recommend to use - Dockerfile-<OS distro>-complete to get all the ROCm Math libs installed which are required for PyTorch.
 
-`Dockerfile <https://github.com/ROCmSoftwarePlatform/pytorch/wiki/Dockerfile>`_
+`Dockerfile <https://github.com/RadeonOpenCompute/ROCm-docker/tree/master/dev>`_
```
Comment (jithunnair-amd) [inline Deep_learning/Deep-learning.rst:378]: The first dockerfile also has a PyTorch build step at the end. The latter one is simply to install ROCm afaict.

File: Deep_learning/Deep-learning.rst [hunk 4]
```diff
@@ -407,45 +407,44 @@ Note: This will mount your host home directory on /data in the container.
...
-
-This step is optional but most PyTorch scripts will use torchvision to load models. E.g., running the pytorch examples requires torchvision.
+This step is optional. However, most PyTorch scripts will use torchvision to load models. For example, running the PyTorch examples requires torchvision.
 
 9. Commit the container to preserve the pytorch install (from the host):
 
 ::
 
-  sudo docker commit <container_id> -m 'pytorch installed'
+  sudo docker commit <container_id> -m <new image name>
```
Comment (jithunnair-amd) [inline Deep_learning/Deep-learning.rst:444]: The `-m` flag is for an optional commit message (https://docs.docker.com/engine/reference/commandline/commit/). The image name at the end is optional too, but if we want to keep that optional argument, it should read: `sudo docker commit <container_id> -m 'pytorch installed' <new image name>`, or  `sudo docker commit <container_id> <new image name>`

**Review by jithunnair-amd — COMMENTED** [review]
Probably a bit late, but fwiw...

File: Deep_learning/Deep-learning.rst [hunk 5]
```diff
@@ -251,8 +251,7 @@ Building PyTorch for ROCm
 
 This is a quick guide to setup PyTorch with ROCm support inside a docker container. Assumes a .deb based system. See `ROCm install <https://github.com/RadeonOpenCompute/ROCm#supported-operating-systems---new-operating-systems-available>`_ for supported operating systems and general information on the ROCm software stack.
 
-
-A ROCm install version 3.3 is required currently.
+Note: Currently, ROCm install version 3.3 is required.
```
[the remark this answers is not in the document]
Comment (pruthvistony) [reply Deep_learning/Deep-learning.rst:254]: @Rmalavally ,
Can you please take care of this comment.

File: Deep_learning/Deep-learning.rst (hunk 4 above)
Comment (pruthvistony) [reply Deep_learning/Deep-learning.rst:444]: Its better to revert this line of change will raise it in next PR.

Comment (pruthvistony) [reply Deep_learning/Deep-learning.rst:444]: https://github.com/RadeonOpenCompute/ROCm_Documentation/pull/125

**Pull request merged**